src/utils.test.js

```diff
@@ -8,172 +8,181 @@ import {
-        },
-        {
-          id: 2, description: 'Task 2', completed: false, index: 2,
-        },
-        {
-          id: 3, description: 'Task 3', completed: false, index: 3,
-        },
-      ];
-      const updatedList = moveTaskToBottom(2, initialList);
-      expect(updatedList[2].id).toBe(2);
-      localStorage.setItem('todoList', JSON.stringify(updatedList));
-      expect(JSON.parse(localStorage.getItem('todoList'))).toEqual(updatedList);
-    });
-  });
-});
\ No newline at end of file
+class LocalStorageMock {
+  constructor() {
+    this.store = {};
+  }
+
+  getItem(key) {
+    return this.store[key] || null;
+  }
+
+  setItem(key, value) {
+    this.store[key] = value.toString();
+  }
+
+  removeItem(key) {
+    delete this.store[key];
+  }
+
+  clear() {
+    this.store = {};
+  }
+}
+beforeEach(() => {
+  Object.defineProperty(window, 'localStorage', { value: new LocalStorageMock() });
+});
+test('addTask should add a task to the list', () => {
+  const initialList = [
+    { id: 1, description: 'Task 1', completed: false },
+    { id: 2, description: 'Task 2', completed: true },
+  ];
+  const newTaskDescription = 'New Task';
+  localStorage.setItem('todoList', JSON.stringify(initialList));
+  const updatedList = addTask(newTaskDescription, initialList);
+  const newTask = updatedList.find((task) => task.description === newTaskDescription);
+  expect(newTask).toBeDefined();
+  expect(newTask.completed).toBe(false);
+  expect(updatedList).toHaveLength(3);
+});
+test('addTask should update localStorage with the new list', () => {
+  const initialList = [
+    { id: 1, description: 'Task 1', completed: false },
+    { id: 2, description: 'Task 2', completed: true },
+  ];
+  const newTaskDescription = 'New Task';
+  localStorage.setItem('todoList', JSON.stringify(initialList));
+  const updatedList = addTask(newTaskDescription, initialList);
+  const newTask = updatedList.find((task) => task.description === newTaskDescription);
+  const storedListString = localStorage.getItem('todoList');
+  const storedList = JSON.parse(storedListString);
+  expect(newTask).toBeDefined();
+  expect(storedList).toContainEqual(newTask);
+  expect(newTask.completed).toBe(false);
+  expect(updatedList).toHaveLength(3);
+});
+test('deleteTask should delete a task from the list', () => {
+  const initialList = [
+    { id: 1, description: 'Task 1', completed: false },
+    { id: 2, description: 'Task 2', completed: true },
+    { id: 3, description: 'Task 3', completed: false },
+  ];
+  localStorage.setItem('todoList', JSON.stringify(initialList));
+  const updatedList = deleteTask(2, initialList);
+  const expectedList = [
+    { id: 1, description: 'Task 1', completed: false },
+    { id: 3, description: 'Task 3', completed: false },
+  ];
+  expect(updatedList).toHaveLength(2);
+  expect(updatedList).toEqual(expectedList);
+});
+test('clearCompletedTasks should clear completed tasks from the list', () => {
```

Comment:
👍 Your test cases have surpassed my expectations, showing a high level of attention to detail. However, there's an area that needs improvement:

- To enhance the organization and readability of your test suite, you should group your tests using the [describe()](https://jestjs.io/docs/api#describename-fn) method. By creating logical blocks using the `describe` blocks, you can group related test cases together, making it easier to understand the test suite's structure. Grouping tests into descriptive blocks allows for better management and maintenance, especially when you need to focus on specific methods or classes. Following this approach aligns with the project requirements and will improve the overall clarity of your test suite.

Keep up the great work with your test cases, and I'm confident that organizing them using `describe` blocks will take your test suite to the next level! 🌟🧪🚀

Reply:
Did it, thanks!!!
![image](https://github.com/lerfast/To-do-list-webpack/assets/111823606/d74f3227-cde7-4fca-b11a-34a6f702b7aa)